crates/telio-dns/src/nameserver.rs

```diff
@@ -254,18 +260,40 @@ impl LocalNameServer {
             return Err(String::from("Invalid IpPacket"));
         }
 
+        let payload = match ip.next_level_protocol() {
+            IpNextHeaderProtocols::Udp => {
+                let udp_request = UdpPacket::new(ip.payload())
+                    .ok_or_else(|| String::from("Failed to build UdpPacket from request packet"))?;
+                LocalNameServer::process_udp_packet(&udp_request, |udp_packet| {
+                    ip.udp_checksum(udp_packet)
+                })?
+            }
+            IpNextHeaderProtocols::Tcp => {
+                let tcp_request = TcpPacket::new(ip.payload())
+                    .ok_or_else(|| String::from("Failed to build TcpPacket from request packet"))?;
+                PayloadRequestInfo::Tcp {
+                    source_port: tcp_request.get_source(),
+                    destination_port: tcp_request.get_destination(),
+                    next_sequence: tcp_request.get_sequence() + 1,
```

Comment:
:thinking: So there is interesting edge-case, as this sequence number will only work for TCP-SYN packets, or packets without any TCP data payload. Technically it is not possible to send any data without first establishing connection, which involved TCP SYN packet.

But I could imagine edge-case scenario, where a client was communicating with 100.64.0.2 before enabling DNS, and then telio-DNS gets enabled, and it first receives non-TCP-SYN packet in TCP connection, in which case, TCP RST may get ignored later. 

But this is going into so edge-casy territory, that probably adding all the extra complexity to handle this situation is net-negative instead of simply waiting until client-side application times-out. :thinking: